### ISSUE TITLE
Map is always blank (white) when using MbTiles

### ISSUE THREAD

**Issue by nadavpld** [pauldendulk/Mapsui#179]
Hi, 
I was trying to get mapsui work with the local tile server but couldnt make it work properly. The map is always blank.
I worked alongside the samples but got a behaviour similar to #106  from 04.04.
I tried the sample mbtiles you guys attached and the world_countries.mbtiles from arcgis.

Would appreciate if you can point me.
This is the code I gathered and tried running from the few samples I found : 

            MapControl mapcontrol = FindViewById<MapControl>(Resource.Id.mapcontrol);
 
            var path = System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal);
 
            path = Path.Combine(path, "world_countries.mbtiles");
 
            string dbFileName = "world_countries.mbtiles";
 
            string dbFilePath = Android.App.Application.Context.GetDatabasePath(dbFileName).AbsolutePath;
 
            string databaseFolder = System.IO.Path.GetDirectoryName(dbFilePath);
 
            if(!File.Exists(dbFilePath))
            {
                Stream input = Assets.Open(dbFileName);
                Stream output = new FileStream(dbFilePath, FileMode.OpenOrCreate);
                byte[] buffer = new byte[1024];
                int b = buffer.Length;
                int length;
                while((length=input.Read(buffer, 0 , b))>0)
                {
                    output.Write(buffer, 0, length);
                }
                output.Flush();
                output.Close();
                input.Close();
            }
 
            var mbTilesTileSource = new MbTilesTileSource(new SQLiteConnectionString(dbFilePath, true));
           
            mapcontrol.Map.Layers.Add(new TileLayer(mbTilesTileSource));

Thank you !

**Comment by pauldendulk** [comment]
I renamed your title to .. MbTiles. 'Tile server' makes me think of an actual web service running on your local machine. If this is what you actually mean please correct me.

**Comment by pauldendulk** [comment]
There is an MbTilesSample.cs that works in WPF. It used the ```torrejon-de-ardoz.mbtiles``` file. The best approach is to first test this in WPF to see how it is expected to work. Then try to get it to work in Android. The modifications you need to make are: 
- Deploy the mbtiles file to device somehow
- Get the path to the file right.

**Comment by pauldendulk** [comment]
oh, and you also need to add a nuget reference to the platform specific sqlite packages. I think you need this one:
https://www.nuget.org/packages/SQLitePCLRaw.bundle_green/

**Comment by charlenni** [comment]
First check, if your stream contains anything. If this is the case, than check your zoom factor. I tried it by my own some time ago and had the same problem. It seems, that you have to select the correct zoom level to see anything.

**Comment by pauldendulk** [comment]
There have been some improvements on MbTiles. If you use MbTilesSample.CreateMap() you should see the same on Android as on WPF.

**Comment by nadavpld** [comment]
@charlenni the mbTilesTileSource do contain something. I have read somewhere else about the zoom and gave it a try.. didnt work unfortunately, but I might did it wrong - would appreciate if you can supply an example.
I tried the following (at the end of the above code) : 
`ZoomHelper.ZoomToBoudingbox(mapcontrol.Map.Viewport, -180, -90, 180m 90, 1100, 620`

@pauldendulk MpTilesSample.CreateMap() was my reference. It couldnt load the mbtiles files as implemented there so I did the copy-to-storage thingy..

Thank you for the quick replies :]

**Comment by pauldendulk** [comment]
MpTilesSample.CreateMap() is the example that you should try to follow. If this does not work, nothing else will work. You have to:
- Deploy the mbtiles file to a location that can be accessed by your application.
- Change the path to that location
- Add bundle_green nuget package: https://www.nuget.org/packages/SQLitePCLRaw.bundle_green/

**Comment by nadavpld** [comment]
This is exactly what I did before asking you guys, I attached the code in the first message.
It might be what @charlenni said, I tried to play a little bit with the zoom level but still getting the blank map.

**Comment by pauldendulk** [comment]
The MbTilesSample has bing maps as the backgroud. Did bing maps show on Android as in the WPF sample?

**Comment by nadavpld** [comment]
I'm working in an offline enviroment so I cant check the bing maps atm (this is the reason for the offline map feature, I will be able to check it from home).
I did notice something wierd in the xaml, I could see it only when moving to the "designer" view - I'm getting a DllNotFoundException on libSkiaSharp.so
Any thoughts ?

**Comment by charlenni** [comment]
That's my code to display an offline MBTiles map:

`			var filePath = DependencyService.Get<IFileHelper>().GetLocalFilePath("TileFile.mbtiles");
			var sqlConnection = new SQLite.SQLiteConnectionString(filePath, false);
			var tileSource = new BruTile.MbTilesTileSource(sqlConnection);
			var tileLayer = new TileLayer(tileSource);

			mapView.Map.Layers.Add(tileLayer);
`

Try to add a 

`	mapView.Map.NavigateTo(tileSource.Extent.ToBoundingBox());
`

**Comment by charlenni** [comment]
Did any of the samples work for you? Even if you don't have an online connection, something should be displayed on screen. Try the random test.

**Comment by pauldendulk** [comment]
Install SkiaSharp.Views
https://www.nuget.org/packages/SkiaSharp.Views

**Comment by nadavpld** [comment]
@pauldendulk I installed it already, got the above error when it was already installed.

@charlenni Are you sure thats valid ? `mapView.Map.NavigateTo(tileSource.Extent.ToBoundingBox());`
I dont have an Extent over there.. 
Got to say that the MbTilesTileSource class in the dll I Installed from nuget looks different from the one available here : 
https://github.com/BruTile/BruTile/blob/master/BruTile.MbTiles.Pcl/MbTilesTileSource.cs

I couldnt get any tiles to show up on my screen, I tried the Bing \ OSM as paul said, I was only getting the credits text below the map to show up, no tiles. I will give it another try when ariving home.
